pkg/controller/storagecluster/cephcluster.go

```diff
@@ -403,8 +403,16 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster, serverVersion *version.
 						in := getPlacement(sc, "osd-prepare-tsc")
 						(&in).DeepCopyInto(&preparePlacement)
 
+						// Eliminating default multi level topology spread constraint in case of host failure domain.
+						if sc.Spec.FlexibleScaling {
+							preparePlacement.TopologySpreadConstraints = preparePlacement.TopologySpreadConstraints[:1]
+						}
+
 						if len(topologyKeyValues) >= replica {
 							preparePlacement.TopologySpreadConstraints[0].TopologyKey = topologyKey
+						} else {
+							// Eliminating default multi level topology spread consrtaints.
+							preparePlacement.TopologySpreadConstraints = preparePlacement.TopologySpreadConstraints[:1]
```

Comment:
If Line 408 executes, what is the value of this line? I don't think this does any change to TSC.

Reply:
Now the changes have been made to add multiple TSC only when zone/rack is detected as failure domain.